samplers.go

```diff
@@ -19,6 +19,7 @@ type DDMetric struct {
 	Tags       []string      `json:"tags,omitempty"`
 	MetricType string        `json:"type"`
 	Hostname   string        `json:"host"`
+	DeviceName string        `json:"device_name"`
```

Comment:
this is supported on metrics? I thought it was only for events (it wouldn't be the first time we used undocumented params, but I'd probably want to try it and make sure it works first)